spring-boot-project/spring-boot-tools/spring-boot-buildpack-platform/src/main/java/org/springframework/boot/buildpack/platform/docker/transport/HttpClientTransport.java

```diff
@@ -146,14 +112,35 @@ private Response execute(HttpUriRequest request) {
 		try {
 			ClassicHttpResponse response = this.client.executeOpen(this.host, request, null);
 			int statusCode = response.getCode();
+
 			if (statusCode >= 400 && statusCode <= 500) {
 				byte[] content = readContent(response);
+				// Always close the response for error paths
 				response.close();
+
+				if (statusCode == 407) {
+					String detail = null;
+					Message json = deserializeMessage(content);
+					if (json != null && StringUtils.hasText(json.getMessage())) {
+						detail = json.getMessage();
+					}
```

Comment:
Why are you handling JSON here? The issue states that the error has "a plain text message".

Reply:
This still doesn't look right to me, I'm afraid. Why would a JSON `Message` be sent back in an error case? If you have evidence of that, please do share it. Otherwise, I think this should just deal with plain text.